cli/src/pcluster/models/compute_fleet_status_manager.py

```diff
@@ -62,7 +62,152 @@ def is_start_status(status):
+            raise
+
+    def update_status(self, request_status, in_progress_status, final_status, wait_transition=False):
+        """
+        Update the status of the compute fleet and wait for a status transition.
+
+        It updates the status of the fleet to request_status and then waits for it to be updated to final_status,
+        by eventually transitioning through in_progress_status
+        """
+        compute_fleet_status = self.get_status()
+        if compute_fleet_status == ComputeFleetStatus.UNKNOWN:
+            raise Exception("Could not retrieve compute fleet status.")
+
+        if compute_fleet_status == final_status:
+            LOGGER.info("Compute fleet already in %s status.", final_status)
+            return
+
+        LOGGER.info("Compute fleet status is: %s. Submitting status change request.", compute_fleet_status)
+        if compute_fleet_status not in {request_status, in_progress_status, final_status}:
+            self._update_status(current_status=compute_fleet_status, next_status=request_status)
+
+        if not wait_transition:
+            LOGGER.info("Request submitted successfully. It might take a while for the transition to complete.")
+            LOGGER.info("Please run 'pcluster status' if you need to check compute fleet status")
+            return
+
+        LOGGER.info("Submitted compute fleet status transition request. Waiting for status update to start...")
+        compute_fleet_status = self._wait_for_status_transition(wait_on_status=request_status, timeout=180)
+        if compute_fleet_status == in_progress_status:
+            LOGGER.info(
+                "Compute fleet status transition is in progress. This operation might take a while to complete..."
+            )
+            compute_fleet_status = self._wait_for_status_transition(wait_on_status=in_progress_status, timeout=600)
+
+        if compute_fleet_status != final_status:
+            raise Exception(
+                "Unexpected final state {} probably due to a concurrent status update request.".format(
+                    compute_fleet_status
+                )
+            )
+        LOGGER.info("Compute fleet status updated successfully.")
+
+    def _wait_for_status_transition(self, wait_on_status, timeout=300, retry_every_seconds=15):
+        current_status = self.get_status()
+        start_time = time.time()
+        while current_status == wait_on_status and not self._timeout_expired(start_time, timeout):
+            current_status = self.get_status()
+            time.sleep(retry_every_seconds)
+
+        if current_status == wait_on_status:
+            raise TimeoutError("Timeout expired while waiting for status transition.")
+
+        return current_status
+
+    @staticmethod
+    def _timeout_expired(start_time, timeout):
+        return (time.time() - start_time) > timeout
+
+
+class SlurmComputeFleetStatusManager:  # TODO to be removed after refactoring
```

Comment:
As discussed pcluster CLI needs to be able to manage status update for older versions.